src/flowcean/environments/directory_loader.py

```diff
@@ -0,0 +1,40 @@
+from collections.abc import Callable, Iterable
+from pathlib import Path
+
+from flowcean.core.environment.offline import OfflineEnvironment
+
+LoaderFunction = Callable[[Path], OfflineEnvironment]
+
+
+def load_directory(
+    path: Path | str,
+    load_function: LoaderFunction,
+    *,
+    pattern: str = "*",
+    include_files: bool = True,
+    include_folders: bool = True,
+) -> Iterable[OfflineEnvironment]:
+    """Load environments from a directory.
+
+    This helper function can be used to load environments from multiple files
+    or folders. First all files and folders in the `path` matching the
+    `pattern` are selected. These are then passed to the `load_function`
+    which in turn creates the actual environment.
+
+    Args:
+        path: The path to the directory from which environments are loaded.
+        load_function: Function handle to a function loading an environment
+            from a path and returning it.
+        pattern: A glob pattern. Matching files and folders will be passed to
+            `load_function`.
+        include_files: Specify whether to pass files to the `load_function`.
+        include_folders: Specify whether to pass folders to the
+            `load_function`.
+    """
+    for item in Path(path).glob(pattern):
+        if (item.is_file() and include_files) or (
+            item.is_dir() and include_folders
+        ):
+            environment = load_function(item)
+            if environment is not None:
+                yield environment
```

Comment:
Not sure whether this actually is easier to read/understand, but I am a fan of functional style expressions in this case. Something like:

```python
map(
  load_function,
  filter(
    lambda item: item.is_file() or item.is_dir(),
    Path(path).glob(pattern)
  )
)
```

You can additionally extract the intermediate iterables into variables to increase readability and give it more structure.

Reply:
I would personally keep it in the "long" form - espacially because we can easily add a progress ar here (which I did in #128). If we can add the progressbar to the function style expression as well, I'm all in for it 😄

Reply:
The tqdm just takes an `Iterable` and produces another. So, yes, you can.

Reply:
Then the functional style is fine for me. I updated the code accordingly.